PackageViewModel/PackageSearch/LocalPackageSearcher.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace PackageExplorerViewModel.PackageSearch
+{
+    class LocalPackageSearcher<T> where T : IPackageSearchMetadata
+    {
+        private readonly SearchContext _searchContext;
+
+        public LocalPackageSearcher(SearchContext searchContext)
+        {
+            _searchContext = searchContext;
+        }
+
+        public IEnumerable<T> SearchPackages(IEnumerable<T> packages)
+        {
+            var searchText = _searchContext.SearchText;
+            var searchFilter = _searchContext.Filter;
+
+            if (!searchFilter.IncludePrerelease)
+            {
+                packages = packages.Where(p => p is PackageSearchMetadata meta && !meta.Version.IsPrerelease);
+            }
+
+            if (_searchContext.IsIdSearch)
+            {
+                packages = packages.Where(p => string.Equals(p.Identity.Id, searchText, StringComparison.OrdinalIgnoreCase));
+            }
+            else if (!String.IsNullOrEmpty(searchText))
+            {
+                // Support multiple terms
+                var searchValues = SplitValues(searchText, " ");
+
+                packages = packages
+                    .Select(p => new { score = searchValues.Sum(s => CalcScore(p, s)), package = p })
+                    .Where(it => it.score > 0) // first filtering before sorting for performance reasons
+                    .OrderByDescending(it => it.score)
+                    .Select(it => it.package);
+            }
+
+            return packages;
+        }
+
+        private static int CalcScore(T package, string searchText)
+        {
+            var score = ScoreForEquals(package.Identity.Id, searchText);
```

Comment:
FYI This isn't fully following https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/issues/607#issuecomment-517950958

Reply:
What are the differences?

Reply:
Can the package scoring be parellelized to speed it up? Parallel.ForEach, etc?

Reply:
Maybe, but it's now already blazing fast. (instant experience with 1900 packages). The slow part is already done (download JSON  + parse all JSON).